packages/admin/src/pages/super-admin-dashboard/manage-departments/edit/[id]/index.page.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  FlexibleQuestionPageLayout,
+  QuestionPageGetServerSideProps,
+  TextInput,
+} from 'gap-web-ui';
+import { GetServerSidePropsContext } from 'next';
+import CustomLink from '../../../../../components/custom-link/CustomLink';
+import Meta from '../../../../../components/layout/Meta';
+import getConfig from 'next/config';
+import InferProps from '../../../../../types/InferProps';
+import { getUserTokenFromCookies } from '../../../../../utils/session';
+import {
+  getDepartment,
+  updateDepartmentInformation,
+} from '../../../../../services/SuperAdminService';
+import { Department } from '../../../types';
+
+export function getServerSideProps(context: GetServerSidePropsContext) {
+  const fetchPageData = async (jwt: string) => {
+    const { name: departmentName, ggisID } = await getDepartment(
+      context.params?.id as string,
+      jwt
+    );
+    return {
+      departmentName,
+      ggisID,
+    };
+  };
+
+  const handleRequest = async (body: Omit<Department, 'id'>, jwt: string) =>
+    updateDepartmentInformation(body, context.params?.id as string, jwt);
+
+  return QuestionPageGetServerSideProps({
```

Comment:
pass in your types here, so you dont have to define an extra type on line 45

Reply:
Line 45 is not a type, I am destructuring `departmentName `and `ggisID `from `pageData`. Types for these constants are already provided by `InferProps`